contracts/modules/cover/Cover.sol

```diff
@@ -295,16 +295,18 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon {
     uint coverId,
     BuyCoverParams memory buyCoverParams,
     PoolAllocationRequest[] memory poolAllocations
-  ) external payable onlyMember {
+  ) external payable onlyMember whenNotPaused {
 
     CoverData memory cover = _coverData[coverId];
     uint lastCoverSegmentIndex = _coverSegments[coverId].length - 1;
     CoverSegment memory lastCoverSegment = coverSegments(coverId, lastCoverSegmentIndex);
 
+    require(ICoverNFT(coverNFT).ownerOf(coverId) == msg.sender, "Cover: Only the owner can edit the cover");
```

Comment:
It could be useful to also allow approved addresses to edit for better composability.

Reply:
:+1: